Browser Support

Hi.

Does this library support browsers, or only node js apps?
I am trying to compile it on angular but failing.
'fs' is missing from the 'Tunnel' lib.

> Browsers are supported but this looks like a bug. Adding a sample would ensure it works in angular and doesn't regress going forward.

ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'fs' in 'C:\Projekte\AngularLibrarys\demo\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'http' in 'C:\Projekte\AngularLibrarys\demo\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'http' in 'C:\Projekte\AngularLibrarys\demo\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'https' in 'C:\Projekte\AngularLibrarys\demo\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'https' in 'C:\Projekte\AngularLibrarys\demo\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'net' in 'C:\Projekte\AngularLibrarys\demo\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/Util.js
Module not found: Error: Can't resolve 'path' in 'C:\Projekte\AngularLibrarys\demo\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'tls' in 'C:\Projekte\AngularLibrarys\demo\node_modules\tunnel\lib'
i ｢wdm｣: Failed to compile.

ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'fs' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'http' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'http' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'https' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'https' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'net' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\tunnel\lib'
ERROR in ./node_modules/typed-rest-client/Util.js
Module not found: Error: Can't resolve 'path' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\typed-rest-client'
ERROR in ./node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'tls' in 'C:\Projekte\AngularLibrarys\eceg\node_modules\tunnel\lib'
i ｢wdm｣: Failed to compile.

Hi.
Tried from different pc and a simple Angular Project.

ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'assert' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'events' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'fs' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'http' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'http' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'https' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'https' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'net' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/Util.js
Module not found: Error: Can't resolve 'path' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'tls' in 'D:\Work\Web\demo\node_modules\tunnel\lib'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/HttpClient.js
Module not found: Error: Can't resolve 'url' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/typed-rest-client/Util.js
Module not found: Error: Can't resolve 'url' in 'D:\Work\Web\demo\node_modules\typed-rest-client'
ERROR in D:/Work/Web/demo/node_modules/tunnel/lib/tunnel.js
Module not found: Error: Can't resolve 'util' in 'D:\Work\Web\demo\node_modules\tunnel\lib'

--- a/lib/HttpClient.ts
+++ b/lib/HttpClient.ts
@@ -4,9 +4,9 @@
 import url = require("url");
 import http = require("http");
 import https = require("https");
-import tunnel = require("tunnel");
 import ifm = require('./Interfaces');
-import fs = require('fs');
+let fs: any;
+let tunnel: any;
 
 export enum HttpCodes {
     OK = 200,
@@ -122,17 +122,22 @@ export class HttpClient implements ifm.IHttpClient {
 
             this._certConfig = requestOptions.cert;
 
-            // cache the cert content into memory, so we don't have to read it from disk every time 
-            if (this._certConfig && this._certConfig.caFile && fs.existsSync(this._certConfig.caFile)) {
-                this._ca = fs.readFileSync(this._certConfig.caFile, 'utf8');
-            }
+            if (this._certConfig) {
+                // If using cert, need fs
+                fs = require('fs');
 
-            if (this._certConfig && this._certConfig.certFile && fs.existsSync(this._certConfig.certFile)) {
-                this._cert = fs.readFileSync(this._certConfig.certFile, 'utf8');
-            }
-
-            if (this._certConfig && this._certConfig.keyFile && fs.existsSync(this._certConfig.keyFile)) {
-                this._key = fs.readFileSync(this._certConfig.keyFile, 'utf8');
+                // cache the cert content into memory, so we don't have to read it from disk every time 
+                if (this._certConfig.caFile && fs.existsSync(this._certConfig.caFile)) {
+                    this._ca = fs.readFileSync(this._certConfig.caFile, 'utf8');
+                }
+    
+                if (this._certConfig.certFile && fs.existsSync(this._certConfig.certFile)) {
+                    this._cert = fs.readFileSync(this._certConfig.certFile, 'utf8');
+                }
+    
+                if (this._certConfig.keyFile && fs.existsSync(this._certConfig.keyFile)) {
+                    this._key = fs.readFileSync(this._certConfig.keyFile, 'utf8');
+                }
             }
 
             if (requestOptions.allowRedirects != null) {
@@ -437,7 +442,12 @@ export class HttpClient implements ifm.IHttpClient {
         }
 
         if (useProxy) {
-            const agentOptions: tunnel.TunnelOptions = {
+            // If using proxy, need tunnel
+            if (!tunnel) {
+                tunnel = require('tunnel');
+            }
+
+            const agentOptions = {
                 maxSockets: maxSockets,
                 keepAlive: this._keepAlive,
                 proxy: {

--- a/package.json
+++ b/package.json
@@ -1,6 +1,6 @@
 {
   "name": "typed-rest-client",
-  "version": "1.4.0",
+  "version": "1.5.0",
   "description": "Node Rest and Http Clients for use with TypeScript",
   "main": "./RestClient.js",
   "scripts": {

